src/applications/disability-benefits/all-claims/validations.js

```diff
@@ -374,3 +374,45 @@ export function validateBooleanGroup(
     errors.addError(atLeastOne);
   }
 }
+
+/* Military history validations */
+export const validateAge = (
+  errors,
+  dateString,
+  _formData,
+  _schema,
+  _uiSchema,
+  _currentIndex,
+  appStateData,
+) => {
+  if (moment(dateString).isBefore(moment(appStateData.dob).add(13, 'years'))) {
+    errors.addError('Your start date must be after your 13th birthday');
+  }
+};
+
+// partial matches for reserves
+// NOAA & Public Health Service are considered to be active duty
+const reservesList = ['Reserve', 'National Guard'];
+
+export const validateSeparationDate = (
+  errors,
+  dateString,
+  _formData,
+  _schema,
+  _uiSchema,
+  currentIndex,
+  appStateData,
+) => {
+  const { allowBDD, servicePeriods = [] } = appStateData;
+  const branch = servicePeriods[currentIndex]?.serviceBranch || '';
+  const isReserves = reservesList.some(match => branch.includes(match));
+  if (!allowBDD && !isReserves && moment(dateString).isAfter(moment())) {
+    errors.addError('Your separation date must be in the past');
+  } else if (
+    allowBDD &&
+    !isReserves &&
+    moment(dateString).isAfter(moment().add(180, 'days'))
```

Comment:
Could potentially migrate to date-fns [isAfter](https://date-fns.org/v2.19.0/docs/isAfter) here as well. Possibly (again untested lol) something like: 

```
import isAfter from 'date-fns/isAfter';
import parseISO from 'date-fns/parseISO';
import add from 'date-fns/add'

if(isAfter(parseISO(dateString), add(new Date(), {days: 180}))){...}
```

Reply:
I will in a future iteration 😸